scripts/generate-doc.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh -eu
+
+VERSION="$(git describe --always)"
+PROJ_ROOT="$PWD"
+TMP_DIR="$(mktemp -d)"
+
+cd "$TMP_DIR"
+
+go get -u github.com/skygeario/openapi3-gen/cmd/openapi3-gen
+
+git clone --no-checkout "$PROJ_ROOT" gh-pages
+cd gh-pages
+git checkout -t origin/gh-pages || git checkout --orphan gh-pages
+git reset --hard
+
+mkdir -p apis/auth
+openapi3-gen -dir "$PROJ_ROOT" -output "apis/auth/$VERSION.yaml" ./cmd/auth/... ./pkg/auth/...
+
+URL="https://generator.swagger.io/?url=https://skygeario.github.io/skygear-server/apis/auth/$VERSION.yaml"
+sed "s|{URL}|$URL|g" "$PROJ_ROOT/scripts/template/index.html" > index.html
+
+git add .
+git commit -m "Update documentation for $VERSION" || true
```

Comment:
Why `|| true`?

Reply:
The commit would fail if nothing changed. On a second thought, I should just notify the user about it.